ui/theming/compose/src/main/kotlin/com/edricchan/studybuddy/ui/theming/compose/StudyBuddyTheme.kt

```diff
@@ -0,0 +1,52 @@
+package com.edricchan.studybuddy.ui.theming.compose
+
+import android.content.Context
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.dynamicDarkColorScheme
+import androidx.compose.material3.dynamicLightColorScheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import com.edricchan.studybuddy.ui.theming.compose.theme.StudyBuddyTypography
+import com.edricchan.studybuddy.ui.theming.compose.theme.compat.StudyBuddyCompatDarkColors
+import com.edricchan.studybuddy.ui.theming.compose.theme.compat.StudyBuddyCompatLightColors
+import com.edricchan.studybuddy.ui.theming.compose.theme.m3.StudyBuddyM3DarkColors
+import com.edricchan.studybuddy.ui.theming.compose.theme.m3.StudyBuddyM3LightColors
+import com.edricchan.studybuddy.ui.theming.compose.theme.supportsDynamicColor
+
+/**
+ * Sets the [MaterialTheme] for all of the Composables in [content].
+ *
+ * On Android 12+, the system-wide dynamic colours will be used.
+ * @param context The [Context] to be used to retrieve the dynamic colour (Android 12+).
+ * @param enableDarkTheme Whether dark theme should be enabled.
+ * @param useM3Colors Whether the M3 colours ([StudyBuddyM3LightColors],
+ * [StudyBuddyM3DarkColors]) should be used instead of [StudyBuddyCompatLightColors] and
+ * [StudyBuddyCompatDarkColors] when [dynamic theming][useDynamicTheme] is disabled.
+ * @param useDynamicTheme Whether the system's wallpaper colour should be used. This defaults to
+ * `true` for supported devices (Android 12+), or `false` otherwise.
+ */
+@Composable
+fun StudyBuddyTheme(
+    context: Context = LocalContext.current,
```

Comment:
I don't see why a context is needed as a parameter here, there's nothing that you can pass in that'll actually affect `dynamicDarkColorScheme/dynamicLightColorScheme`, right?